### DISCUSSION TITLE
What does question mark means in relations?

### DISCUSSION THREAD

**Discussion by ziimakc** [prisma/prisma#18293] [Q&A] [answered]
I can't find a good descriptions about `?` usage in relations and what is the behavior in each case.

```
model User {
  userId          String    @id
  userDetails                userDetails?
  // userDetails                userDetails  - what is the difference?
}

model userDetails {
   id          String   @id
  userId    String

  user User @relation(fields: [userId], references: [userId])
  // user User? @relation(fields: [userId], references: [userId]) -  what is the difference?
}
```

**Comment by ludralph** [comment, marked as answer]
Hello @ZiiMakc 👋 

The question mark is used to make a field optional. See this section of the [documentation](https://www.prisma.io/docs/reference/api-reference/prisma-schema-reference#-modifier-1)

In the first data model, the `userDetails` field with the User model with the `?` is optional, meaning it can be null. On the other hand, in the second data model, the `user` field in the userDetails model without the `?` is required and cannot be null.

If the user field is made optional by adding a question mark, as in `user User?, it can be null. This means that there may be userDetails records without corresponding User records.
